fiftyone/utils/eval/segmentation.py

```diff
@@ -490,25 +541,64 @@ def _from_dict(cls, d, samples, config, eval_key, **kwargs):
             eval_key,
             d["pixel_confusion_matrix"],
             d["classes"],
+            weights_dict=_parse_index_dict(d.get("weights_dict", None)),
+            ytrue_ids_dict=_parse_index_dict(d.get("ytrue_ids_dict", None)),
+            ypred_ids_dict=_parse_index_dict(d.get("ypred_ids_dict", None)),
             missing=d.get("missing", None),
             **kwargs,
         )
 
     @staticmethod
-    def _parse_confusion_matrix(confusion_matrix, classes):
+    def _parse_confusion_matrix(
+        confusion_matrix,
+        classes,
+        weights_dict=None,
+        ytrue_ids_dict=None,
+        ypred_ids_dict=None,
+    ):
+        have_ids = ytrue_ids_dict is not None and ypred_ids_dict is not None
```

Comment:
This is a non-trivial amount of data transformation, which, as currently implemented, must be performed every time a `SegmentationResults` object is loaded. May want to consider optimizing the data model here for speed.